frog/server/stream.js

```diff
@@ -1,40 +1,42 @@
 // @flow
 
 import { Meteor } from 'meteor/meteor';
-import { generateReactiveFn, splitPathObject } from 'frog-utils';
+import { generateReactiveFn } from 'frog-utils';
 
+import { Activities } from '/imports/api/activities';
+import { activityTypesObj } from '/imports/activityTypes';
 import { serverConnection } from './share-db-manager';
 import { SharedbCache } from '../imports/api/cache';
 import LearningItem from '../imports/ui/LearningItem';
 
-const safelyInsertObject = (doc, dataFn, path, value, instanceId) => {
-  const { insertObject, insertPath } = splitPathObject(doc.data, path, value);
-  dataFn.objInsert(insertObject, insertPath);
-  dataFn.objInsert(instanceId, [
-    ...path.slice(0, path.length - 1),
-    'instanceId'
-  ]);
-};
-
 Meteor.methods({
-  stream: (activity, instanceId, path, value) => {
+  stream: (activity, instanceId, value) => {
     if (activity.streamTarget) {
+      const target = Activities.findOne(activity.streamTarget);
       const docId = activity.streamTarget + '/all';
+      const mergeFunction = activityTypesObj[target.activityType].mergeFunction;
+      if (!mergeFunction) {
+        return;
+      }
+      const toSend = {
+        data: { '1': { ...value, id: '1' } },
+        config: {}
+      };
       if (SharedbCache[docId]) {
-        const [doc, dataFn] = SharedbCache[docId];
-        safelyInsertObject(doc, dataFn, path, value, instanceId);
+        const [dataFn] = SharedbCache[docId];
+        mergeFunction(toSend, dataFn);
```

Comment:
While testing with Chat, if the mergeFunction crashes, the whole app crashes and the server needs to be restarted. I suggest surrounding this with a try/catch.